code/modules/shuttle/shuttle_events/_shuttle_events.dm

```diff
@@ -0,0 +1,145 @@
+			target_corner = list(bounding_coords[1], bounding_coords[2])
+			spawn_offset = list(-SHUTTLE_TRANSIT_BORDER, 0)
+
+	if(spawning_behaviour & SHUTTLE_EVENT_HIT_SHUTTLE)
+		///so we get either the horizontal width or vertical width, which would both equal the amount of spawn tiles
+		var/tile_amount = abs((direction == NORTH || SOUTH) ? bounding_coords[1] - bounding_coords[3] :  bounding_coords[2] - bounding_coords[4])
+		for(var/i in 0 to tile_amount)
+			var/list/target_coords = list(target_corner[1] + step_dir[1] * i + spawn_offset[1], target_corner[2] + step_dir[2] * i + spawn_offset[2])
+			spawning_turfs_hit.Add(locate(target_coords[1], target_coords[2], port.z))
+	if(spawning_behaviour & SHUTTLE_EVENT_MISS_SHUTTLE)
+		for(var/i in 1 to SHUTTLE_TRANSIT_BORDER)
+			//Get the corner tile, and move away from the shuttle and towards the cordon
+			spawning_turfs_miss.Add(locate(target_corner[1] - step_dir[1] * i + spawn_offset[1], target_corner[2] - step_dir[2] * i + spawn_offset[2], port.z))
+			var/corner_delta = list(bounding_coords[3] - bounding_coords[1], bounding_coords[2] - bounding_coords[4])
+			//Get the corner tile, but jump over the shuttle and then continue unto the cordon
+			spawning_turfs_miss.Add(locate(target_corner[1] + corner_delta[1] * step_dir[1] + step_dir[1] * i + spawn_offset[1], target_corner[2] + corner_delta[2] * step_dir[2] + step_dir[2] * i + spawn_offset[2], port.z))
+
+
+/datum/shuttle_event/simple_spawner/event_process()
+	. = ..()
+
+	if(!.)
+		return FALSE
+
+	if(!LAZYLEN(spawning_list) && self_destruct_when_empty)
+		return SHUTTLE_EVENT_CLEAR
+
+	if(prob(spawn_probability_per_process))
+		for(var/i in 1 to spawns_per_spawn)
+			spawn_movable(get_type_to_spawn())
+
+///Pick a random turf from the valid turfs we got. Overwrite if you need some custom picking
+/datum/shuttle_event/simple_spawner/proc/get_spawn_turf()
+	RETURN_TYPE(/turf)
+	return pick(spawning_turfs_hit + spawning_turfs_miss)
+
+///Spawn stuff! if you're not using this, don't use the simple_spawner subtype
+/datum/shuttle_event/simple_spawner/proc/spawn_movable(spawn_type)
+	post_spawn(new spawn_type (get_spawn_turf()))
+
+///Not tecccccccccccchnically a getter since we can also pop the type out of the spawning list, but thats optional anyway
+/datum/shuttle_event/simple_spawner/proc/get_type_to_spawn()
+	. = pick_weight(spawning_list)
+	if(remove_from_list_when_spawned) //if we have this enabled, we decrease the pickweight by 1 till it runs out
+		spawning_list[.] -= 1
+		if(spawning_list[.] < 1)
+			spawning_list.Remove(.)
+
+///Do any post-spawn edits you need to do
+/datum/shuttle_event/simple_spawner/proc/post_spawn(atom/movable/spawnee)
+	ADD_TRAIT(spawnee, TRAIT_FREE_HYPERSPACE_SOFTCORDON_MOVEMENT, src) //Lets us spawn and move further away from the shuttle without being teleported into space
+	ADD_TRAIT(spawnee, TRAIT_DEL_ON_SPACE_DUMP, src) //if we hit the cordon, we get deleted. If the shuttle can make you, it can qdel you
```

Comment:
```suggestion
	ADD_TRAIT(spawnee, TRAIT_FREE_HYPERSPACE_SOFTCORDON_MOVEMENT, REF(src)) //Lets us spawn and move further away from the shuttle without being teleported into space
	ADD_TRAIT(spawnee, TRAIT_DEL_ON_SPACE_DUMP, REF(src)) //if we hit the cordon, we get deleted. If the shuttle can make you, it can qdel you
```